kong/init.lua

```diff
@@ -811,13 +818,32 @@ function Kong.balancer()
     end
   end
 
+  local pool_opts
+  local kong_conf = kong.configuration
+
+  if enable_keepalive and kong_conf.upstream_keepalive_pool_size > 0
+     and subsystem == "http"
+  then
+    local pool = balancer_data.ip .. "|" .. balancer_data.port
+
+    if balancer_data.scheme == "https" then
+      -- upstream_host is SNI
+      pool = pool .. "|" .. ngx.var.upstream_host
```

Comment:
Should we add the upstream client certificate ID somewhere in here as well?

Reply:
Totally agreed, forgot we support upstream mTLS now :+1:

Reply:
Done, thanks for the suggestion!